kafka-0.10.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala

```diff
@@ -279,6 +287,17 @@ final case class KafkaConsumerConfig(
     "retry.backoff.ms" -> retryBackoffTime.toMillis.toString
   )
 
+  predefinedProperties.keySet
+    .intersect(properties.keySet)
+    .headOption
+    .foreach { duplicatedKey =>
+      throw new BadValue(
```

Comment:
Not a fan of just throwing stuff in this case, maybe we could make `Map` overwrite configs in `case class`?

Reply:
Sure, not a problem, will make my life easier

Reply:
@Avasil actually, I think it should be the other way round - if something is set on the case class it should take precedence as it's being used by monix-kafka to configure producer/consumer - imagine a situation where auto commit on the case class is set to true, but properties override it to false -> that way monix-kafka is going to use different value (true) than the one passed to the underlying client (false)

Reply:
If we go with `Map` having higher priority then we can do something like `map.getOrElse("key", config.key)`

Reply:
@Avasil sure, that works too, it's bit more code as we have to check every property, where in the approach currently implemented in this PR can simply rely on the fact that config will have higher priority and overwrite whatever is passed in properties. In a way that makes more sense to me, but I can see reasons why it also makes sense the other way around. Is your preference to have map with higher priority then?